wildfly-config-gen/src/main/java/org/jboss/provisioning/plugin/wildfly/configgen/WfProvisionedConfigHandler.java

```diff
@@ -207,17 +207,18 @@ private void writeAttributes(ProvisionedFeature feature) throws ProvisioningDesc
         if (elemValue == null) {
             throw new ProvisioningException("Required element " + WfConstants.ADDR_PARAMS + " is missing for " + spec.getId());
         }
-        List<String> addrParams  = null;
-        try {
-            addrParams = parseList(annotation.getElementAsList(WfConstants.ADDR_PARAMS), annotation.getElementAsList(WfConstants.ADDR_PARAMS_MAPPING));
-        } catch (ProvisioningDescriptionException e) {
-            throw new ProvisioningDescriptionException("Saw an empty parameter name in annotation " + WfConstants.ADDR_PARAMS + "="
-                    + elemValue + " of " + spec.getId());
-        }
-        if (addrParams == null) {
-            return Collections.emptyList();
+        List<String> addrParams = Collections.emptyList();
+        if (!"server-root".equals(elemValue)) {
+            try {
+                addrParams = parseList(annotation.getElementAsList(WfConstants.ADDR_PARAMS), annotation.getElementAsList(WfConstants.ADDR_PARAMS_MAPPING));
+            } catch (ProvisioningDescriptionException e) {
+                throw new ProvisioningDescriptionException("Saw an empty parameter name in annotation " + WfConstants.ADDR_PARAMS + "="
+                        + elemValue + " of " + spec.getId());
+            }
+            if (addrParams == null) {
```

Comment:
This will never be null actually. I see it is actually present in the current code as well.